yearn/treasury/treasury.py

```diff
@@ -177,90 +188,122 @@ def maker_collateral(self, block=None) -> dict:
-            'MakerDAO': self.maker_debt(block=block),
-        }
-        if not block or block >= 11315910:
-            debt['Unit.xyz'] = self.unit_debt(block=block)
-        # TODO: self.accounts_payable()
+        debt = {address: {} for address in self.addresses}
+
+        maker_debt = self.maker_debt(block=block)
+        if maker_debt:
+            for address, data in maker_debt.items():
+                debt[address].update(data)
+        
+        unit_debt = self.unit_debt(block=block)
+        if unit_debt:
+            for address, data in unit_debt.items():
+                debt[address].update(data)
+
+        compound_debt = self.compound_debt(block=block)
+        if compound_debt:
+            for address, data in compound_debt.items():
+                debt[address].update(data)
+
+        debt = {key: value for key, value in debt.items() if len(value)}
         return debt
 
-    def accounts_payable(self, block=None) -> dict:
-        for i, partner in enumerate(partners):
-            if i == 1:
-                flat_wrappers = []
-                for wrapper in partner.wrappers:
-                    if isinstance(wrapper, Wrapper):
-                        flat_wrappers.append(wrapper)
-                    elif isinstance(wrapper, WildcardWrapper):
-                        flat_wrappers.extend(wrapper.unwrap())
-                for wrapper in flat_wrappers:
-                    print(wrapper.protocol_fees(block=block))
-
     def maker_debt(self, block=None) -> dict:
         proxy_registry = contract('0x4678f0a6958e4D2Bc4F1BAF7Bc52E8F3564f3fE4')
         cdp_manager = contract('0x5ef30b9986345249bc32d8928B7ee64DE9435E39')
-        # ychad = contract('ychad.eth')
-        ychad = contract('0xfeb4acf3df3cdea7399794d0869ef76a6efaff52')
         vat = contract('0x35D1b3F3D7966A1DFe207aa4514C12a259A0492B')
-        proxy = proxy_registry.proxies(ychad)
-        cdp = cdp_manager.first(proxy)
-        urn = cdp_manager.urns(cdp)
         ilk = encode_single('bytes32', b'YFI-A')
-        art = vat.urns(ilk, urn, block_identifier=block).dict()["art"]
-        rate = vat.ilks(ilk, block_identifier=block).dict()["rate"]
-        debt = art * rate / 1e45
         dai = '0x6B175474E89094C44Da98b954EedeAC495271d0F'
-        debt = {dai: {'balance': debt, 'usd value': debt}}
-        return debt
+        maker_debt = {}
+        for address in self.addresses:
+            proxy = proxy_registry.proxies(address)
+            cdp = cdp_manager.first(proxy)
+            urn = cdp_manager.urns(cdp)
+            art = vat.urns(ilk, urn, block_identifier=block).dict()["art"]
+            rate = vat.ilks(ilk, block_identifier=block).dict()["rate"]
+            debt = art * rate / 1e45
+            maker_debt[address] = {dai: {'balance': debt, 'usd value': debt}}
+        return maker_debt
 
     def unit_debt(self, block=None) -> dict:
+        # NOTE: This only works for YFI based debt, must extend before using for other collaterals
```

Comment:
hmm, I'm not sure what we could assert there, since you don't pass a particular collateral type as an arg. Maybe better to change the method name for clarity?